### ISSUE TITLE
REP13: DHCP Config Generator (hiring task)

### ISSUE THREAD

**Issue by mkurek** [allegro/ralph#2208]
# DHCP config file generator for DCAssets objects

Materials:
- http://linux.die.net/man/5/dhcpd.conf
- https://www.centos.org/docs/5/html/Deployment_Guide-en-US/s1-dhcp-configuring-server.html
# Task

Create simple Django view which serves dhcp.conf plain text contents

Result: 2 URL end points:
- endpoint serving DHCP "entries" (in a text format)
- endpoint serving DHCP "networks" config (in a text format)

These two URLs will be in the end fetched and merged into single dhcp.conf file on the DHCP remote side. It should be possible to do it  for example using simple curl script(with 2 URLsinvocations).

Requirements:
- allow some URL parameters, to generate only entries/networks from specified: DC name, network and environment (think about handling multiple dc or environments, but never do both at the same time - when DC is specified, environment can not be specified)
- output only networks which have 'Broadcast in DHCP configuration' field selected
- MAC address should be stored in one, unified format
# Architecture

Code should land in a separate 'ralph_ext_dhcpgen' python package, so you can install it as optional module. 

Some additional models are necessary to introduce in Ralph package:

(Suggested) Additional models:
`DNSServer` with `ip_address` field (FK to `IPAddress`)
`DHCPEntry` - with `ip` (FK to `IPAddress)` and `Mac` field 

Other models needed are already present in ralph.data_center.models.network
# Example output

Example (single) network config:

```
shared-network "my_network" {
    subnet 10.80.90.0 netmask 255.255.255.0 {
        option routers 10.80.90.1;
        option domain-name "net.com";
        option domain-name-servers 10.20.30.41,10.20.30.42;
        deny unknown-clients;
    }
}
```

**Comment by ar4s** [comment]
What about (i)PXE support? Results of my little research:

In `dhcpd.conf` config file:

```
if exists user-class and option user-class = "iPXE" {
      filename "boot.ipxe";
} else {
      filename "ipxe.pxe";
}
```

For subnet config:

```
subnet 192.168.200.0 netmask 255.255.255.0 {
    range 192.168.200.10 192.168.200.100;
    next-server {{ tftp_server }};
}
```

**Comment by vi4m** [comment]
Let's give a shot, @m-nako !

**Comment by mkurek** [comment]
Solved in #2357
